DQM/SiPixelPhase1Track/plugins/SiPixelPhase1TrackClusters.cc

```diff
@@ -99,10 +110,29 @@ namespace {
     pixelClusterShapeCacheToken_ =
         consumes<SiPixelClusterShapeCache>(iConfig.getParameter<edm::InputTag>("clusterShapeCache"));
 
-    trackerTopoToken_ = esConsumes<TrackerTopology, TrackerTopologyRcd>();
-    trackerGeomToken_ = esConsumes<TrackerGeometry, TrackerDigiGeometryRecord>();
+    trackerTopoToken_ = esConsumes<TrackerTopology, TrackerTopologyRcd, edm::Transition::BeginRun>();
+    trackerGeomToken_ = esConsumes<TrackerGeometry, TrackerDigiGeometryRecord, edm::Transition::BeginRun>();
     clusterShapeHitFilterToken_ =
         esConsumes<ClusterShapeHitFilter, CkfComponentsRecord>(edm::ESInputTag("", "ClusterShapeHitFilter"));
+    templateDBobjectToken_ =
+        esConsumes<SiPixelTemplateDBObject, SiPixelTemplateDBObjectESProducerRcd, edm::Transition::BeginRun>();
+  }
+
+  void SiPixelPhase1TrackClusters::dqmBeginRun(const edm::Run& iRun, const edm::EventSetup& iSetup) {
+    // get geometry
+    edm::ESHandle<TrackerGeometry> tracker = iSetup.getHandle(trackerGeomToken_);
+    assert(tracker.isValid());
+
+    edm::ESHandle<TrackerTopology> tTopoHandle = iSetup.getHandle(trackerTopoToken_);
+    tkTpl = tTopoHandle.product();
+
+    // Initialize 1D templates
+    edm::ESHandle<SiPixelTemplateDBObject> templateDBobject = iSetup.getHandle(templateDBobjectToken_);
+    templateDBobject_ = templateDBobject.product();
```

Comment:
```suggestion
    templateDBobject_ =   iSetup.getData(templateDBobjectToken_);
```
this could be simplified.